src/custom/state/claim/hooks/index.ts

```diff
@@ -244,12 +244,16 @@ export function useUserClaims(account: Account, optionalChainId?: SupportedChain
   const chainId = optionalChainId || connectedChain
 
   const [claimInfo, setClaimInfo] = useState<{ [account: string]: UserClaims | null }>({})
-  const [isLoading, setIsLoading] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
 
   // We'll have claims on multiple networks
   const claimKey = chainId && account && `${chainId}:${account}`
 
   useEffect(() => {
+    if (chainId && !account) {
```

Comment:
This conditional is a bit weird

Why disable loading when `chainId` is set and `account` is not set?

Reply:
Because since the loeader is true by default we need a way do detect when there is no connected wallet and set it the loader to false otherwise it will show infinite loader. But since the account is undefined by default until that hooks runs and checks it, we also don't want to set loader to false on that first check, so that is why we also check for chainId which in the end is always present. So if there is chainId and no account that means that the there is no wallet and at that point we can set loader to false. So this way you don't have the content change and you don't have the infinite loader problem. If you have some better idea how to solve this let me know.

Reply:
Great explanation, thank you :)

I would think that chainId can also be undefined, but if it solves the problem that I'm good